sms_notifservice.py

```diff
@@ -108,23 +104,43 @@ def _prepareNotifQueue(self, notif_queue):
             user_ids = [user_id for user_id in user_ids if user_id]
 
         if len(user_ids) > 0:
-            users = self._user_trackor.get_users_by_ids(user_ids)
+            users = dict((user["userId"], user) for user in self._user_trackor.get_users_by_ids(user_ids))
 
             for notif_queue_rec in notif_queue:
-                for user in users:
-                    tid = user["trackorId"]
-                    if notif_queue_rec.userId == user["userId"]:
-                        notif_queue_rec.trackor_id = tid
-                        try:
-                            notif_queue_rec.phone_number = self._user_trackor.get_phone_number_by_field_name_and_trackor_id(
-                                self._phone_number_field_name,
-                                tid)
-                        except Exception as e:
-                            self._integrationLog.add(LogLevel.ERROR, "Can't get Phone Number from User Trackor. Notif "
-                                                                     "Queue ID = [{}]".format(str(notif_queue_rec.notifQueueId)), str(e))
+                if notif_queue_rec.userId is not None:
+                    notif_queue_rec.phone_number = self._get_phone_number(users[notif_queue_rec.userId])
 
         return notif_queue
 
+    def _get_phone_number(self, user):
+        if self._phone_number_field_name:
+            try:
+                phone_number = self._user_trackor.get_phone_number_by_field_name_and_trackor_id(
+                    self._phone_number_field_name,
+                    user["trackorId"])
+            except Exception as e:
+                self._integrationLog.add(LogLevel.ERROR, "Can't get Phone Number from User Trackor. "
+                                                         "Trackor ID = [{}]".format(str(user["trackorId"])),
+                                        str(e))
+        else:
+            phone_number = user["phoneNumber"]
+
+        return phone_number
+    
+    def _format_phone_number(self, phone_number):
+        original_phone_number = phone_number
+        phone_number = original_phone_number.replace('-', '')
+        phone_number = phone_number.replace('(', '').replace(')', '')
+        if phone_number[0] != '+':
+            phone_number = "+1" + phone_number
+
+        if re.fullmatch('\+\d{11}', phone_number) is None:
```

Comment:
Why ```{11}```? Max length of the phone number is 15. Right?

Reply:
I made a restriction from 2 to 15 digits in the number, because the aws documentation says only about the E.164 format, in this format I saw information only about the maximum value.